gcc-15: build-mpsolve fails with " error: cannot use keyword 'false' as enumeration constant"

On up to date Gentoo x86_64, in cmake build I am getting
```
$ ninja build-libraries
[1/3] Performing build step for 'build-mpsolve'
FAILED: [code=2] libraries/mpsolve/src/build-mpsolve-stamp/build-mpsolve-build /home/dima/software/M2/M2/BUILD/build/libraries/mpsolve/src/build-mpsolve-stamp/build-mpsolve-build 
cd /home/dima/software/M2/M2/BUILD/build/libraries/mpsolve/build && LD_LIBRARY_PATH=/home/dima/software/M2/M2/BUILD/build/usr-host/lib: /usr/bin/gmake -j4 -C src/libmps && LD_LIBRARY_PATH=/home/dima/software/M2/M2/BUILD/build/usr-host/lib: /usr/bin/gmake -j4 -C include && /usr/bin/cmake -E touch /home/dima/software/M2/M2/BUILD/build/libraries/mpsolve/src/build-mpsolve-stamp/build-mpsolve-build
gmake: Entering directory '/home/dima/software/M2/M2/BUILD/build/libraries/mpsolve/build/src/libmps'
/usr/bin/gmake  all-am
gmake[1]: Entering directory '/home/dima/software/M2/M2/BUILD/build/libraries/mpsolve/build/src/libmps'
  CC       chebyshev/chebyshev-evaluation.lo
  CC       chebyshev/chebyshev.lo
  CC       chebyshev/chebyshev-parser.lo
  CC       common/aberth.lo
In file included from ../../include/mps/mps.h:60,
                 from chebyshev/chebyshev-parser.c:11:
../../include/mps/types.h:10:3: error: cannot use keyword 'false' as enumeration constant
   10 | { false = 0, true = 1 } mps_boolean;
      |   ^~~~~
../../include/mps/types.h:10:3: note: 'false' is a keyword with '-std=c23' onwards
In file included from ../../include/mps/mps.h:60,
                 from chebyshev/chebyshev-evaluation.c:11:
../../include/mps/types.h:10:3: error: cannot use keyword 'false' as enumeration constant
   10 | { false = 0, true = 1 } mps_boolean;
      |   ^~~~~
../../include/mps/types.h:10:3: note: 'false' is a keyword with '-std=c23' onwards
In file included from ../../include/mps/mps.h:60,
                 from common/aberth.c:14:
../../include/mps/types.h:10:3: error: cannot use keyword 'false' as enumeration constant
   10 | { false = 0, true = 1 } mps_boolean;
      |   ^~~~~
../../include/mps/types.h:10:3: note: 'false' is a keyword with '-std=c23' onwards
In file included from ../../include/mps/mps.h:60,
                 from chebyshev/chebyshev.c:11:
../../include/mps/types.h:10:3: error: cannot use keyword 'false' as enumeration constant
   10 | { false = 0, true = 1 } mps_boolean;
      |   ^~~~~
../../include/mps/types.h:10:3: note: 'false' is a keyword with '-std=c23' onwards
gmake[1]: *** [Makefile:1506: chebyshev/chebyshev-parser.lo] Error 1
gmake[1]: *** Waiting for unfinished jobs....
gmake[1]: *** [Makefile:1506: chebyshev/chebyshev-evaluation.lo] Error 1
gmake[1]: *** [Makefile:1506: chebyshev/chebyshev.lo] Error 1
gmake[1]: *** [Makefile:1506: common/aberth.lo] Error 1
gmake[1]: Leaving directory '/home/dima/software/M2/M2/BUILD/build/libraries/mpsolve/build/src/libmps'
gmake: *** [Makefile:856: all] Error 2
gmake: Leaving directory '/home/dima/software/M2/M2/BUILD/build/libraries/mpsolve/build/src/libmps'
ninja: build stopped: subcommand failed.
```

--- a/.github/workflows/test_build.yml
+++ b/.github/workflows/test_build.yml
@@ -67,7 +67,7 @@ jobs:
           brew update
           brew tap macaulay2/tap
           brew install --overwrite python
-          brew install automake boost tbb ccache ctags texinfo llvm make ninja yasm libffi msolve googletest fplll eigen
+          brew install automake bison boost tbb ccache ctags texinfo llvm make ninja yasm libffi msolve googletest fplll eigen
           brew install --only-dependencies macaulay2/tap/M2
           brew link factory --force
 
@@ -107,6 +107,7 @@ jobs:
                echo `brew --prefix llvm`/bin              >> $GITHUB_PATH
                PATH=`brew --prefix llvm`/bin:$PATH
                echo `brew --prefix libtool`/libexec/gnubin >> $GITHUB_PATH
+               echo `brew --prefix bison`/bin             >> $GITHUB_PATH
           fi
           # Necessary for clang to find the right libomp
           echo "LIBRARY_PATH=`llvm-config --libdir`" >> $GITHUB_ENV

--- a/M2/cmake/build-libraries.cmake
+++ b/M2/cmake/build-libraries.cmake
@@ -524,6 +524,11 @@ ExternalProject_Add(build-mpsolve
   DOWNLOAD_DIR      ${CMAKE_SOURCE_DIR}/BUILD/tarfiles
   BUILD_IN_SOURCE   ON
   CONFIGURE_COMMAND autoreconf -vif
+            # yacc-generated files shipped in 3.2.3 tarball result in
+            # errors when using newer compilers, so we remove them so
+            # bison can regenerate them
+            COMMAND rm src/libmps/monomial/yacc-parser.c
+            COMMAND rm src/libmps/monomial/yacc-parser.h
             COMMAND ${CONFIGURE} --prefix=${M2_HOST_PREFIX}
                       #-C --cache-file=${CONFIGURE_CACHE}
                       ${shared_setting}

--- a/M2/cmake/build-libraries.cmake
+++ b/M2/cmake/build-libraries.cmake
@@ -517,13 +517,12 @@ _ADD_COMPONENT_DEPENDENCY(programs msolve "gmp;mpfr;flint" MSOLVE_FOUND)
 # https://numpi.dm.unipi.it/software/mpsolve
 # Known issue: tests don't work with static library
 ExternalProject_Add(build-mpsolve
-  URL               ${M2_SOURCE_URL}/mpsolve-3.2.1.tar.gz
-  URL_HASH          SHA256=3d11428ae9ab2e020f24cabfbcd9e4d9b22ec572cf70af0d44fe8dae1d51e78e
+  URL               https://numpi.dm.unipi.it/wp-content/uploads/2025/08/mpsolve-3.2.3.tar.bz2
+  URL_HASH          SHA256=1f2e239c698c783b63a5e6903e76316c0335a01d71c466a8551e8a3f790b3971
   PREFIX            libraries/mpsolve
   SOURCE_DIR        libraries/mpsolve/build
   DOWNLOAD_DIR      ${CMAKE_SOURCE_DIR}/BUILD/tarfiles
   BUILD_IN_SOURCE   ON
-  PATCH_COMMAND     patch --batch -p1 < ${CMAKE_SOURCE_DIR}/libraries/mpsolve/patch-3.2.1
   CONFIGURE_COMMAND autoreconf -vif
             COMMAND ${CONFIGURE} --prefix=${M2_HOST_PREFIX}
                       #-C --cache-file=${CONFIGURE_CACHE}

--- a/M2/libraries/mpsolve/Makefile.in
+++ b/M2/libraries/mpsolve/Makefile.in
@@ -1,14 +1,11 @@
 VPATH = @srcdir@
 HOMEPAGE = https://numpi.dm.unipi.it/software/mpsolve 
-VERSION = 3.2.1
-URL = https://macaulay2.com/Downloads/OtherSourceCode
+VERSION = 3.2.3
+TARFILE = $(LIBNAME)-$(VERSION).tar.bz2
+TAROPTIONS = --bzip2
+URL = https://numpi.dm.unipi.it/wp-content/uploads/2025/08
 
-# obtained from https://numpi.dm.unipi.it/_media/software/mpsolve
-#               https://numpi.dm.unipi.it/_media/software/mpsolve/mpsolve-3.2.1.tar.gz
-
-# the patch disables the python tests, which don't know that the name of the
-# sharable library in Mac OS is libmps.3.dylib instead of libmps.so.3
-PATCHFILE = @abs_srcdir@/patch-$(VERSION)
+# PATCHFILE = @abs_srcdir@/patch-$(VERSION)
 
 LICENSEFILES = README
 

--- a/M2/libraries/mpsolve/Makefile.in
+++ b/M2/libraries/mpsolve/Makefile.in
@@ -12,7 +12,11 @@ LICENSEFILES = README
 # their use of libtool prevents this from working -- somehow "make install" rebuilds mpsolve during the installation, after we strip it
 # STRIPFILES = src/mpsolve/mpsolve
 
-PRECONFIGURE = autoreconf -fvi
+# yacc-generated files shipped in 3.2.3 tarball result in errors when using
+# newer compilers, so we remove them so bison can regenerate them
+PRECONFIGURE = autoreconf -fvi && \
+	rm src/libmps/monomial/yacc-parser.c && \
+	rm src/libmps/monomial/yacc-parser.h
 
 INSTALLTARGET = install-strip
 CONFIGOPTIONS += --disable-shared		\

